app/src/main/java/io/gnosis/safe/ui/dialogs/EnsInputDialog.kt

```diff
@@ -124,8 +138,12 @@ class EnsInputDialog : BaseViewBindingDialogFragment<DialogEnsInputBinding>() {
             binding.successViews.visible(false)
             binding.dialogEnsInputUrlLayout.isErrorEnabled = false
             if (text.toString().isNotEmpty()) {
-                binding.dialogEnsInputProgress.visible(true)
-                job = onUrlChanged(text.toString())
+                if (text.toString().startsWith(" ")) {
+                    binding.dialogEnsInputUrl.setText(text.toString().trim())
+                } else {
```

Comment:
Whay do we not assign the job  if the Url starts with a " "? 
Why not call trim on any input? 
What if the last character is a " "?

Reply:
- ens resolution fails if address is surrounded by spaces.
- we cannot trip on any input as it will lead to infinite cycle.
- with newest change trailing spaces will also be trimmed.

Reply:
actually trimming trailing spaces is not necessary as spaces could be part of the ens name. See https://docs.ens.domains/frequently-asked-questions#what-about-foreign-characters-what-about-upper-case-letters-is-any-unicode-character-valid

Reply:
Why would trimming everything lead to an infinite loop? 
Doesn't the section in the docs mean we should apply nameprep to the input before sending it to the resolver?

Reply:
The weird thing now is that leading spaces are deleted automatically removed and trailing spaces show an error.